Runtime/Interaction/Logic/HPUIGestureLogic.cs

```diff
@@ -219,19 +239,39 @@ public void ComputeInteraction(IHPUIInteractor interactor, IDictionary<IHPUIInte
                 return;
             }
 
-            if (updateTrackingInteractable)
+            KeyValuePair<IHPUIInteractable, HPUIInteractionState> interactableDataToTrack = trackingInteractables
+                .Where(kvp => kvp.Value.Active)
+                .OrderBy(kvp => kvp.Value.CurrentHeuristicValue)
+                .First();
+
+            float heuristicRatio =0;
+            if (currentTrackingInteractable != null)
             {
-                KeyValuePair<IHPUIInteractable, HPUIInteractionState> interactableDataToTrack = trackingInteractables
-                    .Where(kvp => kvp.Value.Active)
-                    .OrderBy(kvp => kvp.Value.CurrentHeuristicValue)
-                    .First();
+                heuristicRatio = (interactableDataToTrack.Value.CurrentHeuristicValue /
+                                  currentTrackingInteractableHeuristic);
+            }
+
+            if (interactableDataToTrack.Key != currentTrackingInteractable)
+            {
+                if (!updateTrackingInteractable)
+                {
+                    if (heuristicRatio < switchCurrentTrackingInteractableThreshold)
```

Comment:
the block computing the `heuristicRatio` should be moved here. The initial value of the `heuristicRatio`  should be infinite.